### PR TITLE
FIX: Data downloading failures from _uncompress_file

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Currently master has data fetching failures. See for instance below link
https://circleci.com/gh/nilearn/nilearn/2280?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

Hopefully, this fix will address the failure of data fetching in master which will therefore fix documentation errors in CircleCI. We have to wait and see if everything goes green.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
There is a CircleCI failure which is related to a warning under sphinx : load_surf_mesh is not in the reference documentation. Could you fix that, to try to have CircleCI green.

**Comment by KamalakerDadi** [comment]
>Could you provide details about how it is supposed to fix a problem?

Download fails with msdl and related which has .zip files with the logic introduced from PR #1372 . Delete an archive file and adjusting for filename after path split will break at conditioning tarfile at ```if tarfile.is_tarfile(file_):```. Since tarfile condition will check for any extension in the ```file_``` which is not because we delete that.

>For me, disabling this logic is wrong.

Could tell me why is it wrong ?

**Comment by AlexandreAbraham** [comment]
It is wrong because you fix a missing file by not deleting it whereas deleting is an explicit functionality of this method. One option is to roll back the changes that introduced the bug or, given that file deletion is now in the zip case, you could just exit the function at the end of this code block. But it is not possible to let gargabe files all around the users hard drive.

**Comment by KamalakerDadi** [comment]
That is what my attempts are in this PR. Rolling back changes as much as possible and carefully looking also not to fail with cobre fetcher. COBRE was failing because after unzipping, the file is deleted again at the end having deleting functionality before ```if not processed``` condition.


>given that file deletion is now in the zip case, you could just exit the function at the end of this code >block. But it is not possible to let gargabe files all around the users hard drive.

I am missing something to understand your second option. In particular, hard drive comment.

**Comment by AlexandreAbraham** [comment]
With your patch, zip archives are no more deleted. This is not acceptable.

**Comment by KamalakerDadi** [comment]
@AlexandreAbraham  @GaelVaroquaux Any more reviews or merge ?

File: nilearn/datasets/utils.py [hunk 1]
```diff
@@ -357,6 +354,10 @@ def _uncompress_file(file_, delete_archive=True, verbose=1):
         if not processed:
             raise IOError(
                     "[Uncompress] unknown archive file format: %s" % file_)
+        else:
+            if delete_archive and os.path.isfile(file_):
+                os.remove(file_)
+
```
Comment (AlexandreAbraham) [inline nilearn/datasets/utils.py]: Remove this.

File: nilearn/datasets/utils.py [hunk 2]
```diff
@@ -330,9 +330,6 @@ def _uncompress_file(file_, delete_archive=True, verbose=1):
             z = zipfile.ZipFile(file_)
             z.extractall(path=data_dir)
             z.close()
-            if delete_archive:
-                os.remove(file_)
-            file_ = filename
```
Comment (AlexandreAbraham) [inline nilearn/datasets/utils.py]: Put this back.

File: nilearn/datasets/utils.py (hunk 1 above)
Comment (AlexandreAbraham) [inline nilearn/datasets/utils.py]: + change condition in line 351
`if os.path.is_file(file_) and tarfile.is_tarfile(file_):`

**Comment by GaelVaroquaux** [comment]
Thanks a lot for your feedback, @AlexandreAbraham . Is the current version satisfactory?

**Comment by AlexandreAbraham** [comment]
Go!

**Comment by GaelVaroquaux** [comment]
Gone!

Thanks @AlexandreAbraham

**Pull request merged**